integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/ui/convert_currency/ConvertView.kt

```diff
@@ -0,0 +1,172 @@
+            CryptoConvertItem.ConvertItemClickListener {
+            override fun onConvertItemClickListener() {
+                if (!dashToCrypto) {
+                    onCurrencyChooserClicked?.invoke()
+                }
+            }
+        }
+        binding.convertToBtn.convertItemClickListener = object :
+            CryptoConvertItem.ConvertItemClickListener {
+            override fun onConvertItemClickListener() {
+                if (dashToCrypto) {
+                    onCurrencyChooserClicked?.invoke()
+                }
+            }
+        }
+    }
+
+    private fun updateUiWithSwap() {
+        setConvertFromBtnData()
+        setConvertToBtnData()
+    }
+
+    private fun setConvertFromBtnData() {
+        binding.convertFromBtn.setCryptoItemArrowVisibility(!dashToCrypto)
+        if (dashToCrypto) {
+            binding.convertFromBtn.setCryptoItemGroupVisibility(true)
+            binding.convertFromBtn.setConvertItemServiceName(R.string.dash_wallet_name)
+            binding.convertFromBtn.setConvertItemTitle(R.string.dash)
+            ContextCompat.getDrawable(context, R.drawable.ic_dash_blue_filled)
+                ?.let { binding.convertFromBtn.setConvertItemIcon(it) }
+        } else {
+            setFromBtnData()
+        }
+    }
+
+    private fun setConvertToBtnData() {
+        binding.convertToBtn.setCryptoItemArrowVisibility(dashToCrypto)
+        if (!dashToCrypto) {
+            binding.convertToBtn.setCryptoItemGroupVisibility(true)
+            binding.convertToBtn.setConvertItemServiceName(R.string.dash_wallet_name)
+            binding.convertToBtn.setConvertItemTitle(R.string.dash)
+            ContextCompat.getDrawable(context, R.drawable.ic_dash_blue_filled)
+                ?.let { binding.convertToBtn.setConvertItemIcon(it) }
+        } else {
+            setToBtnData()
+        }
+    }
+
+    @SuppressLint("SetTextI18n")
+    private fun setFromBtnData() {
+        binding.convertFromBtn.setCryptoItemGroupVisibility(input != null)
+        binding.convertFromDashBalance.isVisible = (input != null)
+        binding.convertFromDashFiatAmount.isVisible = (input != null)
+        input?.let {
+            binding.convertFromBtn.setConvertItemServiceName(it.cryptoWalletService)
+            binding.convertFromBtn.setConvertItemTitle(it.cryptoWalletName)
+            binding.convertFromBtn.setConvertItemIcon(it.icon)
+
+            exchangeRate?.let { currentExchangeRate ->
+
+                val balance = it.balance.toBigDecimal().setScale(8, RoundingMode.HALF_UP).toString()
+                val coin = try {
+                    Coin.parseCoin(balance)
+                } catch (x: Exception) {
+                    Coin.ZERO
+                }
+
+                binding.convertFromDashBalance.text = "${context.getString(R.string.balance)} ${dashFormat.minDecimals(0)
+                    .optionalDecimals(0,8).format(coin)} ${input?.currency}"
```

Comment:
This will ignore localization rules for the currency in front/after the amount. `fiatToString` method would be better suited here.